### PR TITLE
[Issue 3063] Sync from state alone

### PR DESCRIPTION
**Pull request by mbaxter**
## PR Description
Drop initial block argument, and accept initial weak subjectivity state alone.  

Update logic that currently assumes the latest finalized block will be available:
* `RocksDbDatabase`
  * When building the `Store`: 
      * Don't require the latest finalized block to be present
      * If the initial block is missing, pull block metadata from the initial state
  * Update finalization logic to tolerate missing finalized block if it is the initial block
* `WeakSubjectivityInitializer`
  * When validating the initial state CLI argument against an existing database:
      * If the corresponding block is unavailable, try to look up the corresponding state for verification
      * If both the corresponding state and block are missing, just log a warning indicating the initial state argument couldn't be verified against the local chain data
* ForkChoice
  * When initializing at startup, process all available states without requiring the associated block to be available
* State regeneration
  * Look up base states available from the `Store` even if the corresponding block is unavailable
* `PeerChainValidator`
  * Validate peers against local finalized `AnchorPoint` which is kept in memory, avoiding a look up of the latest finalized block

## Fixed Issue(s)
Part of #3063 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: ethereum/core/src/main/java/tech/pegasys/teku/core/stategenerator/StateRegenerationBaseSelector.java [hunk 1]
```diff
@@ -121,13 +121,20 @@ private boolean isBetterThanCurrentRebasedStartingPoint(
...
+              if (maybeBlock.isPresent()) {
+                return maybeBlock.map(
                     block ->
-                        StateAndBlockSummary.create(block, closestAvailableFromStore.getState())));
+                        StateAndBlockSummary.create(block, closestAvailableFromStore.getState()));
+              } else {
+                return Optional.of(
+                    StateAndBlockSummary.create(closestAvailableFromStore.getState()));
+              }
+            });
```
Comment (ajsutton) [inline ethereum/core/src/main/java/tech/pegasys/teku/core/stategenerator/StateRegenerationBaseSelector.java:137]: If we don't actually need the block, would it be better to always create the summary purely off of the state and avoid looking up via blockProvider entirely?  I worry that if the block is sometimes available and sometimes not, it creates more corner cases that could be hard to test.  Whereas if it's never present we know the downstream code has to be able to handle it.

**Review by ajsutton — APPROVED** [review]
LGTM.

File: ethereum/core/src/main/java/tech/pegasys/teku/core/stategenerator/StateRegenerationBaseSelector.java (hunk 1 above)
Comment (mbaxter) [reply ethereum/core/src/main/java/tech/pegasys/teku/core/stategenerator/StateRegenerationBaseSelector.java:137]: Good point - I'll follow up on this suggestion in another PR.  I wanted to avoid changing the current production behavior in this PR, but it does make sense to simplify where we can.

Comment (mbaxter) [reply ethereum/core/src/main/java/tech/pegasys/teku/core/stategenerator/StateRegenerationBaseSelector.java:137]: Added a follow-up ticket here to make sure I don't forget: https://github.com/ConsenSys/teku/issues/3184
